docs/design/iam_containerized.md

```diff
@@ -0,0 +1,262 @@
+- user2 can delete this bucket (after it is empty): although user2 is not the creator, without a bucket policy his root account is the owner so it can delete the bucket.
+
+### Root Accounts Manager
+The root accounts cannot be created using the IAM APIs in containerized deployment.
+
+## Supported Actions and their Request Parameters
+### Supported IAM User Operations
+- IAM CreateUser: Path, UserName (not supported: PermissionsBoundary, Tags.member.N)
+- IAM GetUser: UserName
+- IAM UpdateUser: NewPath, NewUserName, UserName
+- IAM DeleteUser: UserName
+- IAM ListUsers: PathPrefix (not supported: Marker, MaxItems)
+
+### Supported IAM Access Keys Operations
+- IAM CreateAccessKey: UserName
+- IAM GetAccessKeyLastUsed: AccessKeyId
+- IAM UpdateAccessKey: AccessKeyId, Status, UserName
+- IAM DeleteAccessKey: AccessKeyId, UserName
+- IAM ListAccessKeys: UserName (not supported: Marker, MaxItems)
+
+### Supported IAM User Tag Operations
+- IAM TagUser: UserName, Tags.member.N
+- IAM UntagUser: UserName, TagKeys.member.N
+- IAM ListUserTags: UserName
+
+### Supported IAM User Inline Policy Operations
+- PutUserPolicy:  UserName, PolicyDocument, PolicyName
+- DeleteUserPolicy: UserName, PolicyName
+- GetUserPolicy: UserName, PolicyName
+- ListUserPolicies: UserName
+
+### Other
+Would always return an empty list (to check that the user exists it runs GetUser)
+- IAM ListGroupsForUser
+- IAM ListAttachedUserPolicies
+- IAM ListMFADevices
+- IAM ListServiceSpecificCredentials
+- IAM ListSigningCertificates
+- IAM ListSSHPublicKeys
+Would always return an empty list
+- IAM ListAccountAliases
+- IAM ListAttachedGroupPolicies
+- IAM ListAttachedRolePolicies
+- IAM ListGroupPolicies
+- IAM ListGroups
+- IAM ListInstanceProfiles
+- IAM ListOpenIDConnectProviders
+- IAM ListPolicies
+- IAM ListRoles
+- IAM ListSAMLProviders
+- IAM ListServerCertificates
+- IAM ListVirtualMFADevices
+Would always return `NoSuchEntity` error
+- IAM ListEntitiesForPolicy
+- IAM ListInstanceProfilesForRole
+- IAM ListInstanceProfileTags
+- IAM ListMFADeviceTags
+- IAM ListOpenIDConnectProviderTags
+- IAM ListPolicyTags
+- IAM ListPolicyVersions
+- IAM ListRoleTags
+- IAM ListServerCertificateTags
+
+
+#### Naming Scope
+- Account names are unique between the accounts, for example, if we have account with name John, you cannot create a new account with the name John (and also cannot update the name of an existing account to John).
+- Usernames are unique only inside the account, for example: username Robert can be under account-1, and another user with username Robert can be under account-2.
+Note: The username cannot be the same as the account, for example: under account John we cannot create a username John (and also cannot update the name of an existing username to John). The reason for limiting it is that in the IAM API of Access Key (for example ListAccessKeys) it can be done by IAM user on himself or by root account on another user, and it passes the `--user-name` flag.
```

Comment:
As mentioned, I think we need to remove the part of "The reason for limiting..."
The reason for limiting is that NooBaa behaves as AWS IAM.
```suggestion
Note: The username cannot be the same as the account, for example: under account John we cannot create a username John (and also cannot update the name of an existing username to John).
```

You can still add somewhere the information about the flag that was in that sentence.